src/Native/System.Security.Cryptography.Native/pal_ssl.cpp

```diff
@@ -170,35 +170,45 @@ extern "C" int32_t CryptoNative_SslSessionReused(SSL* ssl)
     return SSL_session_reused(ssl) == 1;
 }
 
+static bool StringSpanEquals(const char* lhs, const char* rhs, size_t lhsLength)
+{
+    if (lhsLength != strlen(rhs))
```

Comment:
I'm not usually a fan of macros, but I'm just wondering if we would want to use a macro to get the algorithm name and sizeof(thealgorithmname), to avoid multiple O(n) strlen calls.

Or, alternatively, maybe we just want to check strlen after strncmp?